tools/gulp/docs/index.js

```diff
@@ -82,12 +92,6 @@ module.exports = (gulp, shared) => {
    * @return {Promise<Array>}
    */
   function generateDocPages(pages) {
-    // It's important to require the generatePage method here since the
-    // react-doc.json data file is a dependency. This require() would fail if
-    // it was called before the file existed.
-    // TODO(sawyer): Would it be better if we passed in the relevant React
-    // documentation as a prop, rather than pulling it from the JSON file?
```

Comment:
ℹ️ Answer: yes, this is way cleaner and removes a lot of hacky stuff we were doing. A React component's documentation is now passed in as props (`reactComponentDocs`, `reactExampleSource`, etc). The biggest benefit here is it relates a lot of code bloat.